### PR TITLE
React to hosting changes

### PR DESCRIPTION
**Pull request by HaoK**
MVC changes for https://github.com/aspnet/Hosting/pull/210

cc @davidfowl

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs [hunk 1]
```diff
@@ -114,7 +117,10 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
                 applicationBasePath,
                 applicationWebSiteName);
             services.AddInstance<IApplicationEnvironment>(environment);
-            services.AddInstance<IHostingEnvironment>(new HostingEnvironment(environment));
+            var hostingEnvironment = new HostingEnvironment();
```
Comment (HaoK) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: Maybe we should consider a helper to initialize all 3 of these like the ctor used to with appEnv, hostingEnv.Initialize(applicaitonBasePath) that will set WebRoot/FileProvider?

Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: probably a good idea.  that should avoid the ugly use here of `Microsoft.AspNet.Hosting.Internal` and `HostingUtilities`.

**Comment by dougbu** [comment]
:watch: for a simpler functional test setup

**Comment by HaoK** [comment]
@dougbu @rynowak FYI looks like you should be able to simplify TestHelper further once this set of hosting changes go in, I think you can probably get rid of your own instead of HostingEnvironment since its fully get/settable now.  Also with IStartupFilter, you probably can automatically inject the culture replacer as part of test setup instead of having 450 instances of that line of code :)

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs [hunk 2]
```diff
@@ -114,7 +114,9 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
                 applicationBasePath,
                 applicationWebSiteName);
             services.AddInstance<IApplicationEnvironment>(environment);
```
Comment (HaoK) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs]: Perhaps replacing the application environment might no longer be needed now that hosting enviroment doesn't depend on it

**Pull request closed without merging**